sysutils/pfSense-pkg-apcupsd/files/usr/local/www/widgets/widgets/apcupsd.widget.php

```diff
@@ -0,0 +1,407 @@
+	function compose_apc_contents($widgetkey) {
+		global $user_settings;
+		
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_temp_dis_type"])) {
+			$user_settings["widgets"][$widgetkey]["apc_temp_dis_type"] = "both_deg";
+		}		
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_host_dis"])) {
+			$user_settings["widgets"][$widgetkey]["apc_host_dis"] = "no";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_temp_dis_type_var"])) {
+			$user_settings["widgets"][$widgetkey]["apc_temp_dis_type_var"] = "1";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_load_warning_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_load_warning_threshold"] = "75";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_load_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_load_critical_threshold"] = "90";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_temp_warning_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_temp_warning_threshold"] = "27";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_temp_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_temp_critical_threshold"] = "40";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_charge_warning_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_charge_warning_threshold"] = "50";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_charge_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_charge_critical_threshold"] = "15";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_bage_warning_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_bage_warning_threshold"] = "365";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_bage_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_bage_critical_threshold"] = "720";
+		}
+		
+		if (!(include_once "/usr/local/pkg/apcupsd.inc")) {
+			$rtnstr = "";
+			$rtnstr .= "<div class=\"panel panel-warning responsive\"><div class=\"panel-heading\"><h2 class=\"panel-title\">apcupsd not installed...</h2></div>\n";
+			$rtnstr .= "<pre>\n";
+			$rtnstr .= "Please install and configure apcupsd before using this widget.<br />\n";
+			$rtnstr .= "</pre>\n";
+			$rtnstr .= "</div>\n";
+			print($rtnstr);
+			exit(1);
+		}
+		
+		if (check_nis_running_apcupsd()) {
+			$nisip = (check_nis_ip_apcupsd() != ''? check_nis_ip_apcupsd() : "localhost");
+			$nisport = (check_nis_port_apcupsd() != '' ? check_nis_port_apcupsd() : "3551");
+
+			$ph = popen("/usr/local/sbin/apcaccess -h " . escapeshellarg($nisip) . ":" . escapeshellarg($nisport) . " 2>&1", "r" );
+			while ($v = fgets($ph)) {
+				$results[trim(explode(': ',$v)[0])]=trim(explode(': ',$v)[1]);
+			}
+			pclose($ph);
+
+			$rtnstr = "";
+
+			$rtnstr .= "<tr><td>Status</td><td colspan=\"3\">\n";
+			
+			if($results != null) {
+				switch ($results['STATUS']) {
+					default:
```

Comment:
Usually `default:` is the last option after all other possible values were already tested